File: LaundryLoop.Api/Models/Booking.cs
using System.ComponentModel.DataAnnotations;

namespace LaundryLoop.Api.Models;

public class Booking
{
    public int Id { get; set; }

    // Generated reference shown to user e.g. FFS-123456
    [Required, MaxLength(20)]
    public string Reference { get; set; } = string.Empty;

    // Location
    [Required, MaxLength(100)]
    public string ApartmentName { get; set; } = string.Empty;

    [Required, MaxLength(150)]
    public string ApartmentAddress { get; set; } = string.Empty;

    [Required, MaxLength(20)]
    public string UnitNumber { get; set; } = string.Empty;

    [MaxLength(50)]
    public string? BuzzerCode { get; set; }

    // Contact
    [Required, MaxLength(20)]
    public string Phone { get; set; } = string.Empty;

    [MaxLength(200)]
    public string? Email { get; set; }

    // Schedule
    [Required, MaxLength(20)]
    public string SlotId { get; set; } = string.Empty;

    [Required, MaxLength(50)]
    public string SlotDay { get; set; } = string.Empty;

    [Required, MaxLength(50)]
    public string SlotPeriod { get; set; } = string.Empty;

    [Required, MaxLength(50)]
    public string PickupTime { get; set; } = string.Empty;

    [Required, MaxLength(100)]
    public string DeliveryTime { get; set; } = string.Empty;

    // Laundry Size
    [Required, MaxLength(200)]
    public string LaundrySize { get; set; } = string.Empty;

    [Required]
    public decimal TotalCost { get; set; }

    [MaxLength(1000)]
    public string? Notes { get; set; }

    // Status
    [Required, MaxLength(30)]
    public string Status { get; set; } = "Pending";

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: LaundryPickupDeliveryPHP/src/Models/Booking.php
<?php

declare(strict_types=1);

namespace LaundryLoop\Models;

class Booking
{
    public int     $id               = 0;
    public string  $reference        = '';
    public string  $apartmentName    = '';
    public string  $apartmentAddress = '';
    public string  $unitNumber       = '';
    public ?string $buzzerCode       = null;
    public string  $phone            = '';
    public ?string $email            = null;
    public string  $slotId           = '';
    public string  $slotDay          = '';
    public string  $slotPeriod       = '';
    public string  $pickupTime       = '';
    public string  $deliveryTime     = '';
    public string  $laundrySize      = '';
    public float   $totalCost        = 0.0;
    public ?string $notes            = null;
    public string  $status           = 'Pending';
    public string  $createdAt        = '';
    public string  $updatedAt        = '';

    /**
     * Hydrate from a PDO row (column names are PascalCase in DB).
     */
    public static function fromRow(array $row): self
    {
        $b                  = new self();
        $b->id               = (int)   ($row['Id']               ?? 0);
        $b->reference        = (string) ($row['Reference']        ?? '');
        $b->apartmentName    = (string) ($row['ApartmentName']    ?? '');
        $b->apartmentAddress = (string) ($row['ApartmentAddress'] ?? '');
        $b->unitNumber       = (string) ($row['UnitNumber']       ?? '');
        $b->buzzerCode       =           $row['BuzzerCode']       ?? null;
        $b->phone            = (string) ($row['Phone']            ?? '');
        $b->email            =           $row['Email']            ?? null;
        $b->slotId           = (string) ($row['SlotId']           ?? '');
        $b->slotDay          = (string) ($row['SlotDay']          ?? '');
        $b->slotPeriod       = (string) ($row['SlotPeriod']       ?? '');
        $b->pickupTime       = (string) ($row['PickupTime']       ?? '');
        $b->deliveryTime     = (string) ($row['DeliveryTime']     ?? '');
        $b->laundrySize      = (string) ($row['LaundrySize']      ?? '');
        $b->totalCost        = (float)  ($row['TotalCost']        ?? 0.0);
        $b->notes            =           $row['Notes']            ?? null;
        $b->status           = (string) ($row['Status']           ?? 'Pending');
        $b->createdAt        = (string) ($row['CreatedAt']        ?? '');
        $b->updatedAt        = (string) ($row['UpdatedAt']        ?? '');
        return $b;
    }

    /**
     * Serialize to response array — mirrors MapToResponse() in BookingsController.cs
     */
    public function toResponse(): array
    {
        return [
            'id'               => $this->id,
            'reference'        => $this->reference,
            'status'           => $this->status,
            'createdAt'        => $this->createdAt,
            'apartmentName'    => $this->apartmentName,
            'apartmentAddress' => $this->apartmentAddress,
            'unitNumber'       => $this->unitNumber,
            'buzzerCode'       => $this->buzzerCode,
            'phone'            => $this->phone,
            'email'            => $this->email,
            'slotDay'          => $this->slotDay,
            'slotPeriod'       => $this->slotPeriod,
            'pickupTime'       => $this->pickupTime,
            'deliveryTime'     => $this->deliveryTime,
            'laundrySize'      => $this->laundrySize,
            'totalCost'        => $this->totalCost,
            'notes'            => $this->notes,
        ];
    }
}
